partiql-planner/src/main/kotlin/org/partiql/planner/internal/ir/Nodes.kt

```diff
@@ -393,142 +393,117 @@ internal data class Rex(
             }
         }
 
-        internal data class Path(
-            @JvmField
-            internal val root: Rex,
-            @JvmField
-            internal val steps: List<Step>,
-        ) : Op() {
-            internal override val children: List<PlanNode> by lazy {
-                val kids = mutableListOf<PlanNode?>()
-                kids.add(root)
-                kids.addAll(steps)
-                kids.filterNotNull()
+        public sealed class Path : Op() {
```

Comment:
nit: for the internal plan code (under `planner/internal/ir/*`), `public` is used. It shouldn't matter since the top-level classes are `internal`, but it can be confusing. Would suggest removing the `public` or adding back the explicit `internal` visibility modifiers.